src/pyelliptic/eccblind.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+"""
+ECC blind signature functionality based on "An Efficient Blind Signature Scheme
+Based on the Elliptic CurveDiscrete Logarithm Problem" by Morteza Nikooghadama
+<mnikooghadam@sbu.ac.ir> and Ali Zakerolhosseini <a-zaker@sbu.ac.ir>,
+http://www.isecure-journal.com/article_39171_47f9ec605dd3918c2793565ec21fcd7a.pdf
+"""
+
+# variable names are based on the math in the paper, so they don't conform
+# to PEP8
+# pylint: disable=invalid-name
+
+from .openssl import OpenSSL
+
+
+class ECCBlind(object):  # pylint: disable=too-many-instance-attributes
+    """
+    Class for ECC blind signature functionality
+    """
+
+    # init
+    k = None
+    R = None
+    keypair = None
+    F = None
+    Q = None
+    a = None
+    b = None
+    c = None
+    binv = None
+    r = None
+    m = None
+    m_ = None
+    s_ = None
+    signature = None
+
+    @staticmethod
+    def ec_get_random(group, ctx):
+        """
+        Random point from finite field
+        """
+        order = OpenSSL.BN_new()
+        OpenSSL.EC_GROUP_get_order(group, order, ctx)
+        OpenSSL.BN_rand(order, OpenSSL.BN_num_bits(order), 0, 0)
+        return order
+
+    @staticmethod
+    def ec_invert(group, a, ctx):
+        """
+        ECC inversion
+        """
+        order = OpenSSL.BN_new()
+        OpenSSL.EC_GROUP_get_order(group, order, ctx)
+        inverse = OpenSSL.BN_mod_inverse(0, a, order, ctx)
+        return inverse
+
+    @staticmethod
+    def ec_gen_keypair(group, ctx):
+        """
+        Generate an ECC keypair
+        """
+        d = ECCBlind.ec_get_random(group, ctx)
```

Comment:
d = self.ec_get_random(group, ctx) ?

Reply:
pylint/flake8 complained.

Reply:
because of `@staticmethod`. well, maybe this really should be a static method